src/react-vite/08-stateful-hooks/stateful-hooks.md

```diff
@@ -2,51 +2,117 @@
+> statements that might change the component's behavior. Place hooks at the top
+> of a component's body.
+
+### useState
+
+We can store state that persists through component rendering with the `useState`
+hook. You can set the initial state value when the component **first** renders
+by providing the value as an argument to the hook. _If you do not provide a
+value the initial state value will be `undefined`._
+
+This example shows a `useState` hook being set with an initial value from an
+`initialText` prop.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
+  const [value, setValue] = useState(initialText)
+}
+```
+
+As you can see in the previous example, `useState` returns an array with two
+elements: the first is the current value of the hook, and the second is a setter
+function that is used to update the state value. In the following code, the
+value and setter are being used to update changes to an input element.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
+  const [value, setValue] = useState(initialText)
+  return <input onChange={(evt) => setValue(evt.target.value)} value={value} /> 
+}
+```
+
+Every time a `useState`'s setter is invoked with a new value, React compares the
+new value with the current value. If the values are the same, nothing happens;
+**if the values are different, React will rerender the component** so the new
+value can be used to update the component. In the example above, when the user
+types or changes the value of the input, the `FancyInput` component is rendered
+again, and the input is updated with the current value.
+
+> Watch out! To determine if a state value has changed React compares value
+> types like `object` and array by instance, not by their contents. Other types,
+> like number, string, and boolean, are compared by their value.
+
+### useId
+
+One of the newer hooks that React includes is `useId`. Since the value of every
+`id` attribute in an HTML document must be unique, this hook is useful in
+creating a unique identifier string that can be used as the value of `id` props.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
+  const [value, setValue] = useState(initialText)
+  const id = useId();
+  return (
+    <label htmlFor={id}>Name:</label>
+    <input id={id} onChange={(evt) => setValue(evt.target.value)} value={value} />
+  )
+}
+```
 
-## Objective 1
-
-TODO
-
-### Key concepts
+### Setup
 
-TODO
+<span style="color:red">I think we should add the code related to the state
+object arrays, form, and the selects during setup. Then the exercise will
+involve adding the hooks and wiring state and the setters up to the code.</span>
 
-#### Concept 1
+### Verify
 
-TODO
+These tests will pass when the solution has been implemented properly.
 
-#### Concept 2
+✏️ Update **src/pages/RestaurantList/RestaurantList.test.tsx** to import
+`@testing-library/user-event`:
```

Comment:
These Verify sections always have the attendees copy the whole thing, so you don’t need to highlight just part of the import or anything. Show the `@diff` and they’ll copy the whole file into their local setup.

Reply:
done